src/lib/components/healthselector/Healthselector.component.js

```diff
@@ -0,0 +1,324 @@
+  white-space: nowrap;
+  cursor: pointer;
+  justify-content: space-around;
+
+  ${css`
+    background-color: ${getThemePropSelector('primary')};
+    color: ${getThemePropSelector('textPrimary')};
+    &:hover {
+      background-color: ${getThemePropSelector('primaryDark2')};
+    }
+    &:active {
+      background-color: ${getThemePropSelector('primaryDark2')};
+    }
+  `};
+`;
+
+function Healthselector(props: Props) {
+  const { variant, size, items } = props;
+  const [open, setOpen] = useState(false);
+  const [menuSize, setMenuSize] = useState();
+  const [triggerSize, setTriggerSize] = useState();
+  const selected = items.find((item) => item.selected);
+  const selectedIndex =
+    items.indexOf(selected) >= 0 ? items.indexOf(selected) : 0;
+
+  const refMenuCallback = useCallback((node) => {
+    if (node !== null) {
+      setMenuSize(node.getBoundingClientRect());
+    }
+  }, []);
+
+  const refTriggerCallback = useCallback((node) => {
+    if (node !== null) {
+      setTriggerSize(node.getBoundingClientRect());
+    }
+  }, []);
+
+  const icons = [
+    <ThreeCirclesWrapper>
+      <CSSCircle variant="healthy" />
+      <CSSCircle variant="warning" />
+      <CSSCircle variant="danger" />
+    </ThreeCirclesWrapper>,
+    <CircleWrapper>
+      <CSSCircle variant="healthy" />
+    </CircleWrapper>,
+    <CircleWrapper>
+      <CSSCircle variant="warning" />
+    </CircleWrapper>,
+    <CircleWrapper>
+      <CSSCircle variant="danger" />
+    </CircleWrapper>,
+  ];
+
+  return (
+    <HealthselectorContainer className="sc-healthselector">
+      <TriggerStyled
+        variant={'base'}
+        size={size || 'base'}
+        className="trigger"
+        ref={refTriggerCallback}
+        onBlur={() => setOpen(false)}
+        onClick={(e) => {
+          e.stopPropagation();
+          setOpen(!open);
+        }}
+        opened={open}
+      >
+        <LeftRowWrapper>{icons[selectedIndex]}</LeftRowWrapper>
+        <RightRowWrapper>{(selected && selected.label) || ''}</RightRowWrapper>
+        {open && (
+          <HealthSelectorMenu
+            ref={refMenuCallback}
+            size={menuSize}
+            triggerSize={triggerSize}
+          >
+            <HealthSelectorMenuItem
+              onClick={items && items[0] && items[0].onClick}
+            >
+              <LeftRowWrapper>{icons[0]}</LeftRowWrapper>
+              <RightRowWrapper>
+                {(items[0] && items[0].label) || 'All'}
+              </RightRowWrapper>
+            </HealthSelectorMenuItem>
+            <HealthSelectorMenuItem
+              onClick={items && items[1] && items[1].onClick}
+            >
+              <LeftRowWrapper>{icons[1]}</LeftRowWrapper>
+              <RightRowWrapper>
+                {(items[1] && items[1].label) || 'Ok'}
+              </RightRowWrapper>
+            </HealthSelectorMenuItem>
+            <HealthSelectorMenuItem
+              onClick={items && items[2] && items[2].onClick}
```

Comment:
```suggestion
              onClick={items[2] && items[2].onClick}
```
We actually don't need to check items because we are sure it exists.